cranelift/frontend/src/frontend.rs

```diff
@@ -179,6 +179,38 @@ impl<'short, 'long> InstBuilderBase<'short> for FuncInstBuilder<'short, 'long> {
                 }
             }
 
+            ir::InstructionData::TryCall { exception, .. }
+            | ir::InstructionData::TryCallIndirect { exception, .. } => {
+                let pool = &self.builder.func.dfg.value_lists;
+
+                // Unlike most other jumps/branches and like br_table,
+                // exception tables are capable of having the same successor
+                // appear multiple times, so we must deduplicate.
+                let mut unique = EntitySet::<Block>::new();
+                for dest_block in self
+                    .builder
+                    .func
+                    .stencil
+                    .dfg
+                    .exception_tables
+                    .get(*exception)
+                    .expect("you are referencing an undeclared exception table")
+                    .all_branches()
+                {
+                    let block = dest_block.block(pool);
+                    if !unique.insert(block) {
+                        continue;
+                    }
+
+                    // Call `declare_block_predecessor` instead of `declare_successor` for
+                    // avoiding the borrow checker.
+                    self.builder
+                        .func_ctx
+                        .ssa
+                        .declare_block_predecessor(block, inst);
+                }
+            }
```

Comment:
Mind adding a simple test for a `try_call` where the same successor appears multiple times?